src/Nethermind/Nethermind.Serialization.Rlp/MyTxDecoder/AccessListTxDecoder.cs

```diff
@@ -0,0 +1,243 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Serialization.Rlp.Eip2930;
+
+namespace Nethermind.Serialization.Rlp.MyTxDecoder;
+
+public sealed class AccessListTxDecoder(bool lazyHash = true, Func<Transaction>? transactionFactory = null) : ITxDecoder
+{
+    private static readonly AccessListDecoder AccessListDecoder = new();
+    private readonly bool _lazyHash = lazyHash;
+    private readonly Func<Transaction> _createTransaction = transactionFactory ?? (() => new Transaction());
+
+    public Transaction? Decode(Span<byte> transactionSequence, RlpStream rlpStream, RlpBehaviors rlpBehaviors)
+    {
+        Transaction transaction = _createTransaction();
+        transaction.Type = TxType.AccessList;
+
+        int transactionLength = rlpStream.ReadSequenceLength();
+        int lastCheck = rlpStream.Position + transactionLength;
+
+        DecodePayload(transaction, rlpStream, rlpBehaviors);
+
+        if (rlpStream.Position < lastCheck)
+        {
+            DecodeSignature(transaction, rlpStream, rlpBehaviors);
+        }
+
+        if ((rlpBehaviors & RlpBehaviors.AllowExtraBytes) == 0)
+        {
+            rlpStream.Check(lastCheck);
+        }
+
+        if ((rlpBehaviors & RlpBehaviors.ExcludeHashes) == 0)
+        {
+            if (_lazyHash && transactionSequence.Length <= ITxDecoder.MaxDelayedHashTxnSize)
```

Comment:
I really would like to refactor this to some self contained unit. This would allow us to remove the `lazyHash`, `ITxDecoder.MaxDelayedHashTxnSize` and `transactionSequence`.

The problem lies in `BlobTxDecoder`, where the logic is slightly different.

Reply:
@benaadams